Sources/BraintreeCoreSwift/BTHTTP.swift

```diff
@@ -0,0 +1,542 @@
+import Foundation
+import Security
+
+/// Performs HTTP methods on the Braintree Client API
+// TODO: once BTAPIHTTP + BTGraphQLHTTP are converted this can be internal + more Swift-y
+// TODO: When BTAPIHTTP + BTGraphQL are converted we should update the dictionaries to [String: Any]
+@objcMembers public class BTHTTPSwift: NSObject, NSCopying, URLSessionDelegate {
+// TODO: - Mark interval vs private properties accordingly
+    public typealias RequestCompletion = (BTJSON?, HTTPURLResponse?, Error?) -> Void
+
+    // MARK: - Public Properties
+    /// An array of pinned certificates, each an NSData instance consisting of DER encoded x509 certificates
+    public let pinnedCertificates: [NSData]
+
+    // TODO: Make internal with Swift test?
+    /// Session exposed for testing
+    public lazy var session: URLSession = {
+        let configuration: URLSessionConfiguration = URLSessionConfiguration.ephemeral
+        configuration.httpAdditionalHeaders = defaultHeaders()
+        
+        let delegateQueue: OperationQueue = OperationQueue()
+        delegateQueue.name = "com.braintreepayments.BTHTTP"
+        delegateQueue.maxConcurrentOperationCount = OperationQueue.defaultMaxConcurrentOperationCount
+        
+        return URLSession(configuration: configuration, delegate: self, delegateQueue: delegateQueue)
+    }()
```

Comment:
Made session non-optional by moving its initialization from the `convenience` inits into a lazy var declaration.